File: data/src/main/kotlin/dev/yashgarg/kimai/models/Team.kt
/*
 * Copyright © 2023 Yash Garg.
 * Use of this source code is governed by an MIT-style
 * license that can be found in the LICENSE file or at
 * https://opensource.org/licenses/MIT.
 */
package dev.yashgarg.kimai.models

import androidx.annotation.Keep
import kotlinx.serialization.Serializable

@Keep
@Serializable
data class Team(
  val id: Long,
  val name: String,
  val members: List<Member>? = null,
  val customers: List<ActivityInner>? = null,
  val projects: List<ActivityInner>? = null,
  val activities: List<ActivityInner>? = null,
  val color: String
)

File: lib/data/models/team.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:kimai/data/models/models.dart';

part '../../generated/data/models/team.freezed.dart';
part '../../generated/data/models/team.g.dart';

@freezed
class Team with _$Team {
  const factory Team({
    required int id,
    required String name,
    required List<Member>? members,
    required List<ActivityInner>? customers,
    required List<ActivityInner>? projects,
    required List<ActivityInner>? activities,
    required String color,
  }) = _Team;

  factory Team.fromJson(Map<String, dynamic> json) => _$TeamFromJson(json);
}
